### PR TITLE
beam_integration_benchmark: add beam_it_options, work with k8s

### PR DESCRIPTION
**Pull request by ssisk**
Where work with k8s means:
* can spin up & teardown kubernetes resources defined in a kubernetes yml script
* can read properties from kubernetes resources and use them as options for beam piplines

It also:
* refactors some shared code for interacting with kubernetes out into kubernetes_helper.
* moves beam maven pipline options handling code out into a separate file from the benchmark

cc @asaksena

### CODE REVIEW

**Comment by ssisk** [comment]
also, is there a good benchmark you'd like me to use to make sure the changes to kubernetes_virtual_machine work correctly?

File: perfkitbenchmarker/beam_pipeline_options.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import yaml
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import kubernetes_helper
+
+FLAGS = flags.FLAGS
+
+
+def GetStaticPipelineOptions(options_list):
```
Comment (asaksena) [inline perfkitbenchmarker/beam_pipeline_options.py]: A one line documentation, if possible ? Also I am curious about the options_list data type, so if you have an example feel free to add it to the documentation.

File: perfkitbenchmarker/beam_pipeline_options.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  :param static_pipeline_options: options list passed in via
+      benchmark_spec.dpb_service.static_pipeline_options
+  :param dynamic_pipeline_options: options list passed in via
+      benchmark_spec.dpb_service.dynamic_pipeline_options
+  :return: a list of values of the form "\"--option_name=value\""
+  """
+  # beam_it_options are in [--option=value,--option2=val2] form
+  user_option_list = []
+  if it_options is not None and len(it_options) > 0:
+    user_option_list = it_options.rstrip(']').lstrip('[').split(',')
```
Comment (asaksena) [inline perfkitbenchmarker/beam_pipeline_options.py]: I think long term we can move this to a utility, seems like a useful parsing snippet. Thanks

File: perfkitbenchmarker/beam_pipeline_options.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  # beam_it_args is the old way of passing parameters
+  args_list = []
+  if it_args is not None and len(it_args) > 0:
+    args_list = it_args.split(',')
+
+  return ['"{}"'.format(arg)
+          for arg in args_list + user_option_list + option_list]
+
+
+def ReadPipelineOptionConfigFile():
```
Comment (asaksena) [inline perfkitbenchmarker/beam_pipeline_options.py]: One line documentation, and if you want to add a sample file under the data folder, that will be fine as well, so developers understand the expectation.

File: perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py [hunk 4]
```diff
@@ -58,7 +60,15 @@
 DEFAULT_PYTHON_IT_MODULE = 'apache_beam.examples.wordcount_it_test'
 
 flags.DEFINE_string('beam_it_class', None, 'Path to IT class')
-flags.DEFINE_string('beam_it_args', None, 'Args to provide to the IT')
+flags.DEFINE_string('beam_it_args', None, 'Args to provide to the IT.'
+                    ' Deprecated & replaced by beam_it_options')
+flags.DEFINE_string('beam_it_options', None, 'Pipeline Options sent to the'
+                    ' integration test.')
+flags.DEFINE_string('beam_kubernetes_scripts', None, 'Kubernetes scripts to run'
```
Comment (asaksena) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: Do you think it makes sense to add "Local path to ..." to the flag desc

File: perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py [hunk 5]
```diff
@@ -58,7 +60,15 @@
...
+                    ' which will instantiate a datastore.')
+flags.DEFINE_string('beam_options_config_file', None, 'Yaml file defining'
```
Comment (asaksena) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: same as above

File: perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py [hunk 6]
```diff
@@ -67,31 +77,46 @@ def GetConfig(user_config):
...
-        'specify which sdk is used in the pipeline. For example, java/python.'
-    )
-  if benchmark_spec.dpb_service.service_type != dpb_service.DATAFLOW:
+        'specify which sdk is used in the pipeline. For example, java/python.')
+  if benchmark_config_spec.dpb_service.service_type != dpb_service.DATAFLOW:
     raise NotImplementedError('Currently only works against Dataflow.')
+  if (len(FLAGS.beam_it_options) > 0 and
+      (not FLAGS.beam_it_options.endswith(']') or
+       not FLAGS.beam_it_options.startswith('['))):
+    raise Exception("beam_it_options must be of form"
```
Comment (asaksena) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: Does Beam have any custom exceptions ?

Comment (ssisk) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: I'm not entirely sure what you mean. If there is a correct exception type to be using here, I'd definitely be excited to switch to using it.

**Comment by ssisk** [comment]
Thanks for the review! My dev box is packed for an office move, but I'll make these changes monday.

File: perfkitbenchmarker/beam_pipeline_options.py (hunk 1 above)
Comment (ssisk) [inline perfkitbenchmarker/beam_pipeline_options.py]: done

File: perfkitbenchmarker/beam_pipeline_options.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+  return filledOptions
+
+
+def GenerateAllPipelineOptions(it_args, it_options, static_pipeline_options,
+                               dynamic_pipeline_options):
+  """
+  :param it_args: options list passed in via FLAGS.beam_it_args
+  :param it_options: options list passed in via FLAGS.beam_it_options
+  :param static_pipeline_options: options list passed in via
```
Comment (ssisk) [inline perfkitbenchmarker/beam_pipeline_options.py]: fixed, thanks

File: perfkitbenchmarker/beam_pipeline_options.py (hunk 3 above)
Comment (ssisk) [inline perfkitbenchmarker/beam_pipeline_options.py]: done

File: perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py (hunk 6 above)
Comment (ssisk) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: I have not, can't speak for jason/mark/etc

File: perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py (hunk 4 above)
Comment (ssisk) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: I agree. done

File: perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py (hunk 5 above)
Comment (ssisk) [inline perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py]: Agreed.

**Comment by asaksena** [comment]
LGTM

**Comment by ssisk** [comment]
yay!

**Pull request merged**